cudaextender/src/ungapped_xdrop_kernels.cu

```diff
@@ -183,8 +186,36 @@ __global__ void find_high_scoring_segment_pairs(const int8_t* __restrict__ d_tar
 #pragma unroll
             for (int32_t offset = 1; offset < warp_size; offset = offset << 1)
             {
-                xdrop_done |= __shfl_up_sync(0xFFFFFFFF, xdrop_done, offset);
+                bool temp_xdrop_done = __shfl_up_sync(0xFFFFFFFF, xdrop_done, offset);
+                if (lane_id >= offset)
```

Comment:
Why is this qualifier necessary? `lane_id == 0` will never get these updates. Is that intended?

Reply:
It is a prefix scan with OR operation (just like the earlier loops with AND and MAX operations). Hence, `lane_id == 0` should not be updated